packages/grid_client/scripts/mass_deployments.ts

```diff
@@ -1,119 +1,211 @@
 import {
-  DiskModel,
   FarmFilterOptions,
   FilterOptions,
   generateString,
+  GridClient,
   MachineModel,
   MachinesModel,
   NetworkModel,
-  randomChoice,
+  NodeInfo,
+  TwinDeployment,
 } from "../src";
 import { config, getClient } from "./client_loader";
 import { log } from "./utils";
 
+async function pingNodes(
+  grid3: GridClient,
+  nodes: any[],
+): Promise<Promise<{ node: NodeInfo; error?: Error; res?: unknown }[]>> {
+  const pingPromises = nodes.map(async node => {
+    try {
+      const res = await grid3.zos.pingNode({ nodeId: node.nodeId });
+      return { node, res };
+    } catch (error) {
+      return { node, error };
+    }
+  });
+
+  const results = await Promise.all(pingPromises);
+  return results;
+}
+
 async function main() {
   const grid3 = await getClient();
 
+  // Timeout for deploying vm is 2 min
+  grid3.clientOptions.deploymentTimeoutMinutes = 2;
+  await grid3._connect();
+
   const errors: any = [];
+  const offlineNodes: number[] = [];
   let failedCount = 0;
   let successCount = 0;
+  const batchSize = 5;
+  const totalVMs = 10;
+  const batches = totalVMs / batchSize;
+
+  // resources
+  const cru = 1;
+  const mru = 256;
+  const rootFs = 1;
+  const publicIp = false;
+
+  console.time("Farms Time");
+  const farms = await grid3.capacity.filterFarms({
+    nodeMRU: mru / 1024,
+    nodeSRU: rootFs,
+    publicIp: publicIp,
+    availableFor: await grid3.twins.get_my_twin_id(),
+    randomize: true,
+  } as FarmFilterOptions);
+  console.timeEnd("Farms Time");
+
+  if (farms.length < 1) {
+    throw new Error("No farms found");
+  }
+
+  console.time("Total Deployment Time");
+
+  for (let batch = 0; batch < batches; batch++) {
+    console.time("Batch " + (batch + 1));
+
+    const nodesPromises = Array.from({ length: batchSize }, async () => {
```

Comment:
do we still need to do that? we are already getting all batch nodes with on call